### PR TITLE
[DEP] Tidy up BaseTransformer testing

### PR DESCRIPTION
**Pull request by TonyBagnall**
part of #1446 

- removes some calls to deprecated classes such as BoxCox to reduce the warnings and - removes a duplicate test file for something called OptionalPassthrough.  
- removes the use of ExponentTransformer in testing, replace with MockTransformer that is ExponentTransformer without the offset.

### CODE REVIEW

File: aeon/transformations/tests/test_optional_passthrough.py [hunk 1]
```diff
@@ -1,38 +0,0 @@
-"""Tests for OptionalPassthrough transformer."""
```
Comment (MatthewMiddlehurst) [inline aeon/transformations/tests/test_optional_passthrough.py:1]: What is the other file if this is the duplicate?

File: aeon/testing/estimator_checks.py [hunk 2]
```diff
@@ -60,32 +60,25 @@ def check_estimator(
...
 
-    {'test_clone[ExponentTransformer-0]': 'PASSED',
-    'test_clone[ExponentTransformer-1]': 'PASSED'}
-
-    Running one specific test-fixture-combination for ExponentTransformer
-    >>> check_estimator(
-    ...    ExponentTransformer, fixtures_to_run="test_clone[ExponentTransformer-1]"
-    ... )
-    All tests PASSED!
-    {'test_clone[ExponentTransformer-1]': 'PASSED'}
```
Comment (MatthewMiddlehurst) [inline aeon/testing/estimator_checks.py:88]: why is this gone?

File: aeon/transformations/tests/test_compose.py [hunk 3]
```diff
@@ -1,278 +0,0 @@
-"""Unit tests for transformer composition functionality attached to the base class."""
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/transformations/tests/test_compose.py:1]: why? I thought you wanted rid of the overloading?

File: aeon/transformations/tests/test_optional_passthrough.py (hunk 1 above)
Comment (MatthewMiddlehurst) [reply aeon/transformations/tests/test_optional_passthrough.py:1]: This upload didnt work I think 🙂

File: aeon/transformations/tests/test_compose.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MatthewMiddlehurst) [reply aeon/transformations/tests/test_compose.py:1]: Yes, but its not gone yet. Seems to be more than just overloading there also

File: aeon/transformations/tests/test_optional_passthrough.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/transformations/tests/test_optional_passthrough.py:1]: image

File: aeon/transformations/tests/test_base.py [hunk 4]
```diff
@@ -607,7 +606,7 @@ def test_vectorize_reconstruct_unique_columns():
 def test_numpy_format_outputs():
     """Test that all numpy formats return the same output when converted."""
     X = np.random.random(size=(2, 1, 8))
-    bc = BoxCoxTransformer()
+    bc = Id()
```
[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/transformations/tests/test_base.py:609]: it needs a transpose which I have not implemented for the mock one

[the remark this answers is not in the document]
Comment (TonyBagnall) [reply aeon/transformations/tests/test_base.py:609]: actually thats not true, seems I did but the mock one does not reshape

File: aeon/testing/estimator_checks.py (hunk 2 above)
Comment (TonyBagnall) [reply aeon/testing/estimator_checks.py:88]: just seemed completely excessive to me, but if you insist

**Review by MatthewMiddlehurst — APPROVED** [review]

**Pull request merged**